mmdet/models/builder.py

```diff
@@ -55,5 +56,20 @@ def build_detector(cfg, train_cfg=None, test_cfg=None):
         'train_cfg specified in both outer field and model field '
     assert cfg.get('test_cfg') is None or test_cfg is None, \
         'test_cfg specified in both outer field and model field '
-    return DETECTORS.build(
+
+    rfsearch_cfg = cfg.pop('rfsearch_cfg', None)
+
+    detector = DETECTORS.build(
         cfg, default_args=dict(train_cfg=train_cfg, test_cfg=test_cfg))
+    if rfsearch_cfg is not None:
+        rfsearch_warp = RFSearchHook(
+            mode=rfsearch_cfg.get('mode', 'search'),
+            config=rfsearch_cfg.get('config', None),
+            rfstructure_file=rfsearch_cfg.get('rfstructure_file', None),
+            by_epoch=rfsearch_cfg.get('by_epoch', True),
+            verbose=rfsearch_cfg.get('verbose', True),
+        )
+        detector.init_weights()
```

Comment:
Is `init_weights()` here necessary? mmdetection has initialized weights in `train.py`.

Reply:
I think that it is necessary.
After rfsearch_warp.init_model(detector), the conv.weight will become conv.op_layer.weight.
So we want to init_weights before rfsearch_warp.init_model.